advance_check_void/models/account_payment.py

```diff
@@ -0,0 +1,391 @@
+    is_visible_check = fields.Boolean(
+        "Is Visible Check", help="Use for the visible or invisible check number."
+    )
+    is_readonly_check = fields.Boolean(
+        "Is Readonly Check", help="Use for the readonly or editable check number."
+    )
+    void_reason = fields.Char(string="Void Reason")
+    voided_count = fields.Integer(
+        string="# Voids", compute="_compute_stat_buttons_voided_count"
+    )
+    can_be_voided = fields.Boolean(
+        string="Can be Voided", compute="_compute_can_be_voided"
+    )
+
+    @api.model
+    def _compute_stat_buttons_voided_count(self):
+        for payment in self:
+            res = self.env["payment.check.void"].search_count(
+                [("payment_id", "=", payment.id)]
+            )
+            payment.voided_count = res or 0
+
+    @api.model
+    def _compute_can_be_voided(self):
+        for payment in self:
+            payment.can_be_voided = False
+            if (
+                payment.payment_method_id.payment_type == "outbound"
+                and payment.state == "posted"
+                and payment.payment_state != "reversed"
+            ):
+                if (
+                    payment.payment_method_code == "check_printing"
+                    and payment.is_move_sent
+                ):
+                    payment.can_be_voided = True
+                if payment.payment_method_code == "ACH-Out":
+                    payment.can_be_voided = True
+                if payment.payment_method_code == "manual" and payment.is_move_sent:
+                    payment.can_be_voided = True
+
+    @api.onchange("payment_method_id")
+    def onchange_payment_method(self):
+        """Allow to customer payment to enter check number.
+        - Based on journal check printing manual True/False for vendor."""
+        self.is_visible_check = False
+        self.is_readonly_check = False
+        if self.payment_method_id:
+            if (
+                self.payment_method_id.payment_type == "inbound"
+                and self.payment_method_code in ("check_printing", "ACH-Out", "manual")
+            ):
+                self.is_visible_check = True
+                self.check_number = 0
+            elif self._context.get("is_vendor"):
+                if (
+                    self.payment_method_id.payment_type == "outbound"
+                    and self.payment_method_code
+                    in ("check_printing", "ACH-Out", "manual")
+                ):
+                    self.is_visible_check = False
+            elif (
+                self.payment_method_id.payment_type == "outbound"
+                and self.payment_method_code in ("check_printing", "ACH-Out", "manual")
+            ):
+                self.is_visible_check = True
+                if not self.check_manual_sequencing:
+                    self.is_readonly_check = True
+
+    def print_checks(self):
+        payment_check_void_obj = self.env["payment.check.void"]
```

Comment:
Suggestion to use Object naming conventions, instead of old API 
```suggestion
        PaymentCheckVoid = self.env["payment.check.void"]
```